src/core/lint-hrefs-ids.js

```diff
@@ -0,0 +1,26 @@
+// Module core/lint-hrefs-ids
+// Warns about href's that do not have corresponding id's
+
+import { pub } from "core/pubsubhub";
+
+export const name = "core/lint-hrefs-ids";
+
+export function run(conf, doc, cb) {
+  Array.from(doc.querySelectorAll('a[href]'))
+    .map(a => ({ a, href: a.getAttribute('href') }))
+    .filter(({ href }) => Boolean(href))
+    .filter(({ href }) => href.startsWith('#'))
+    .map(({ a, href }) => ({ a, href, id: href.slice(1) }))
+    .filter(({ id }) => Boolean(id))
+    .filter(({ id }) => !doc.getElementById(id) && !doc.getElementById(decodeURIComponent(id)))
```

Comment:
Not really, the thing is that
```html
<section id='%5B%5BValue%5D%5D'></section>
<a href='#%5B%5BValue%5D%5D'></a>
```
works ok (at least in chromium), that's the reason for two checks (with and without decode) and also for this [test case](https://github.com/w3c/respec/pull/1326/files/975a2376a8a4d492efea2553151a46c234f9689f#diff-ba3e08b1240269a63e384c97af73839aR38).